src/pool/pool_jemalloc.c

```diff
@@ -116,7 +116,15 @@ static bool arena_extent_dalloc(extent_hooks_t *extent_hooks, void *addr,
     (void)committed;    // unused
     (void)arena_ind;    // unused
 
-    return true; // true means failure (unsupported)
+    jemalloc_memory_pool_t *pool = get_pool_by_arena_index(arena_ind);
+
+    umf_result_t ret;
+    ret = umfMemoryProviderFree(pool->provider, addr, size);
```

Comment:
please review which arguments are used and which are not (a few lines above)

Reply:
Done.